src/executor/source/executor/nodes/HttpInputNode.ts

```diff
@@ -9,9 +9,16 @@ export interface HttpInputNode {
  *
  * @param node to execute
  * @param input to be handled
- * @returns the data to execute the next node
+ * @returns the data to execute the next node wraped into an object in fields of the outgoingHandle IDs
  */
 export const executeHttpInputNode = async ( node: Node<HttpInputNode>, input: DataflowInput ): Promise<any> => {
   const dataFieldName = ['GET', 'DELETE'].includes( node.data.params.method?.toUpperCase() || 'GET' ) ? 'query' : 'body'
+  if ( node.data.outgoingHandles.length > 0 ){
+    const results = {}
+    for ( const outgoingHandle of node.data.outgoingHandles ){
+      Object.assign( results, { [outgoingHandle.id]: input[dataFieldName] } )
+    }
+    return results
+  }
   return input[dataFieldName]
```

Comment:
The result has a different form depending on the amount of outgoing handles. This seems wrong.

I'd change the return type of this method to `Promise<Record<string, any>>` – Objects with keys of type `string` (the handle IDs) and `any` content.

If no outgoing handles are present, I'd return an empty object (`{}`) instead of the input – to make it more uniform.

Reply:
I'll change the return type but im not sure about the empty object. If in the future we'd like to display the end result in the editor we would still need the result even if there were no more outgoing handles, id therefore propose to change it to to {['result']: result}.

Reply:
Add functionality only when it's needed. Once we get to implementing this, we might think of a better solution and have to change it anyway.